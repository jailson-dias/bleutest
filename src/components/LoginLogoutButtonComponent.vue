<template>
  <div></div>
</template>

<script>
export default {
  name: "LoginLogoutButtonComponent"
};
</script>