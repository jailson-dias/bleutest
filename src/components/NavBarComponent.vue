<template>
  <div>
    <logo-component />
    <div>
      <menu-button-component />
      <menu-button-component />
      <menu-button-component />
      <menu-button-component />
      <menu-button-component />
    </div>
    <login-logout-button-component />
    <select-language-component />
  </div>
</template>

<script>
import SelectLanguageComponent from "./SelectLanguageComponent";
import LoginLogoutButtonComponent from "./LoginLogoutButtonComponent";
import MenuButtonComponent from "./MenuButtonComponent";
import LogoComponent from "./LogoComponent";

export default {
  name: "NavBarComponent",
  components: {
    MenuButtonComponent,
    LoginLogoutButtonComponent,
    SelectLanguageComponent,
    LogoComponent
  }
};
</script>