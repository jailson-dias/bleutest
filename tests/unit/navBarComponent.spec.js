import { shallowMount } from "@vue/test-utils";

import NavBarComponent from "@/components/NavBarComponent";
import LoginLogoutButtonComponent from "@/components/LoginLogoutButtonComponent";
import SelectLanguageComponent from "@/components/SelectLanguageComponent";

import Vue from "vue";
import Router from "vue-router";

Vue.use(Router);

describe("NavBar Component", () => {
  const build = () => {
    const wrapper = shallowMount(NavBarComponent);

    return {
      wrapper,
      loginLogoutButton: () => wrapper.find(LoginLogoutButtonComponent),
      selectLanguage: () => wrapper.find(SelectLanguageComponent)
    };
  };

  it("É um component do vue", () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Tem o botão de login/logout e a opção de selecionar o idioma", () => {
    const { loginLogoutButton, selectLanguage } = build();

    expect(loginLogoutButton().exists()).toBe(true);
    expect(selectLanguage().exists()).toBe(true);
  });
});
