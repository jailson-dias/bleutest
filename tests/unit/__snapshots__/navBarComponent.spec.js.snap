// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component Ã‰ um component do vue 1`] = `
<div>
  <div>
    <menu-button-component-stub></menu-button-component-stub>
    <menu-button-component-stub></menu-button-component-stub>
    <menu-button-component-stub></menu-button-component-stub>
    <menu-button-component-stub></menu-button-component-stub>
    <menu-button-component-stub></menu-button-component-stub>
  </div>
  <login-logout-button-component-stub></login-logout-button-component-stub>
  <select-language-component-stub></select-language-component-stub>
</div>
`;
